src/services/kuma-api/MockKumaApi.ts

```diff
@@ -1,19 +1,25 @@
+import KumaApi from './KumaApi'
 import { setupHandlers, Mocks } from '@/api/mocks'
 import { setupMockWorker } from '@/api/setupMockWorker'
 import Env from '@/services/env/Env'
-import KumaApi from '@/services/kuma-api/KumaApi'
 
-export default class MockKumaApi extends KumaApi {
-  mocks: Mocks
-  constructor(env: Env, mocks: Mocks) {
-    super(env)
-    this.mocks = mocks
-  }
+export const mockApi = function (parent: typeof KumaApi) {
+  const name = `Mock${parent.name}`
+  const mocked = {
+    [name]: class extends parent {
+      mocks: Mocks
+      constructor(env: Env, mocks: Mocks) {
+        super(env)
+        this.mocks = mocks
+      }
 
-  setBaseUrl(baseUrl: string): void {
-    super.setBaseUrl(baseUrl)
-    if (this.env.var('KUMA_MOCK_API_ENABLED') === 'true') {
-      setupMockWorker(setupHandlers(this.env.var('KUMA_API_URL'), this.mocks))
-    }
+      setBaseUrl(baseUrl: string): void {
+        super.setBaseUrl(baseUrl)
+        if (this.env.var('KUMA_MOCK_API_ENABLED') === 'true') {
+          setupMockWorker(setupHandlers(this.env.var('KUMA_API_URL'), this.mocks))
+        }
+      }
+    },
   }
+  return mocked[name]
```

Comment:
**Question**: Why does the structure of `MockKumaApi` and `DisabledLogger` have to change?

This implementation and that of DisabledLogger are turning increasingly more abstract and hard to follow.

From what I can glimpse, `mockApi` is a function that takes a `KumaApi` class as argument and then returns a class that extends from it. Isn’t this just ... what `MockKumaApi` has been but more complicated?

In short, why is this change needed? Where you call `mockApi` now, why is constructing a new instance of `MockKumaApi` not enough?

Reply:
Because there are times when you have two different classes for example:

```javascript
class LoggerOne {}
class LoggerTwo {}
```

We currently extend this to disable things:

```javascript
class DisabledLogger extends LoggerOne {
// ...stuff
}
class DisabledLogger extends LoggerTwo {
// ...duplicated stuff
}
```

This PR means we can just do:

```javascript
disable(LoggerOne)
disable(LoggerTwo)
```

i.e. we just decorate the classes

Reply:
Where do we currently need to create two independent extensions like that?

Reply:
FYI: This was explained offline

Reply:
I feel like it’s actually an advantage to need to actively disable services in a project rather this being abstracted away, but it seems you’re pretty set on this approach, so I’m not going to argue for it.